runtime/tests/interpreter/for_test.go

```diff
@@ -255,3 +259,108 @@ func TestInterpretForString(t *testing.T) {
 		value,
 	)
 }
+
+type inclusiveRangeForInLoopTest struct {
+	s, e, step        int8
+	expectedLoopCount int
+	onlySignedTest    bool
+}
+
+func TestInclusiveRangeForInLoop(t *testing.T) {
+	t.Parallel()
+
+	baseValueActivation := sema.NewVariableActivation(sema.BaseValueActivation)
+	baseValueActivation.DeclareValue(stdlib.InclusiveRangeConstructorFunction)
+
+	baseActivation := activations.NewActivation(nil, interpreter.BaseActivation)
+	interpreter.Declare(baseActivation, stdlib.InclusiveRangeConstructorFunction)
+
+	testCases := []inclusiveRangeForInLoopTest{
+		{
+			s:                 0,
+			e:                 10,
+			step:              1,
+			expectedLoopCount: 11,
+		},
+		{
+			s:                 0,
+			e:                 10,
+			step:              2,
+			expectedLoopCount: 6,
+		},
+		{
+			s:                 10,
+			e:                 -10,
+			step:              -2,
+			expectedLoopCount: 11,
+			onlySignedTest:    true,
+		},
+	}
+
+	runTestCase := func(t *testing.T, typ sema.Type, testCase inclusiveRangeForInLoopTest) {
+		t.Run(typ.String(), func(t *testing.T) {
+			t.Parallel()
+
+			code := fmt.Sprintf(
+				`
+					fun test(): Int {
+						let s : %[1]s = %[2]d
+						let e : %[1]s = %[3]d
+						let step : %[1]s = %[4]d
+						let r: InclusiveRange<%[1]s> = InclusiveRange(s, e, step: step)
+
+						var count = 0
+						for c in r {
+							count = count + 1
```

Comment:
inside the loop, also use the variable `c` and check if the values assigned to `c` are the correct ones.
e.g: append each `c` to an array and return it from the `test()` function instead of the `count`

Reply:
Updated.